### DISCUSSION TITLE
Clarification on introduction of new parameter in the requestURL in keycloak 26.0.5

### DISCUSSION THREAD

**Discussion by snettar** [keycloak/keycloak#36041] [Q&A] [answered]
We previously had Keycloak 24.0.5 installed on our setup. 
After giving the login credentials of the particular user and hitting the submit button the requestURL is formed something like below

```
Request URL: https://<keycloak-url>/<relative-path>/realms/<realm-name>/login-actions/authenticate?session_code=<some-value>&execution=<some-value>&client_id=<client-id>&tab_id=<some-value>
Request Method: POST
Status Code: 302 Found
```

When we installed Keycloak 26.0.5, and after doing the same operation we are seeing that the requestURL is being formed as something below
```
Request URL: https://<keycloak-url>/<relative-path>/realms/<realm-name>/login-actions/authenticate?session_code=<some-value>&execution=<some-value>&client_id=<client-id>&tab_id=<some-value>&client_data=<some-value> 
Request Method: POST
Status Code: 302 Found
```

We want to know the reason why this new parameter **client_data** is being added? Could not find any such information in the release notes.

**Comment by francescopezzulli** [comment, marked as answer]
Hello, the `client_data` parameter is described [here](https://www.keycloak.org/docs-api/latest/javadocs/org/keycloak/protocol/LoginProtocol.html):

> The purpose of clientData is to be able to send HTTP error response back to the client if authentication fails due some error and authenticationSession is not available anymore (was either expired or removed). So clientData need to contain all the data to be able to send such response. For instance redirect-uri, state in case of OIDC or RelayState in case of SAML etc.

Hope it helps.

**Reply by snettar** [reply]
Thank you @FrancescoPezzulli
